mira/metamodel/templates.py

```diff
@@ -58,6 +70,94 @@ def get_key(self, config: Optional[Config] = None):
             tuple(sorted(self.context.items())),
         )
 
+    def is_equal_to(self, other: "Concept", with_context: bool = False) -> bool:
+        """Test for equality between concepts
+
+        Parameters
+        ----------
+        other :
+            Other Concept to test equality with
+        with_context :
+            If True, do not consider the two Concepts equal unless they also
+            have exactly the same context. If there is no context,
+            ``with_context`` has no effect.
+
+        Returns
+        -------
+        :
+            True if ``other`` is the same Concept as this one
+        """
+        if not isinstance(other, Concept):
+            raise TypeError(f"Cannot check equality between {type(other)} and Concept.")
+
+        # With context
+        if with_context:
+            # Check that the same keys appear in both
+            if set(self.context.keys()) != set(other.context.keys()):
+                return False
+
+            # Check that the values are the same
+            for k1, v1 in self.context.items():
+                if v1 != other.context[k1]:
+                    return False
+
+        # Check that they are grounded to the same identifier
+        if len(self.identifiers) > 0 and len(other.identifiers) > 0:
+            if self.get_curie() != other.get_curie():
+                return False
+            else:
+                pass
+        # If either or both are ungrounded -> can't know -> False
+        else:
+            return False
+
+        return True
+
+    def refinement_of(self, other: "Concept", with_context: bool = False) -> bool:
+        """Check if this Concept is a more detailed version of another
+
+        Parameters
+        ----------
+        other :
+            The other Concept to compare with. Assumed to be less detailed.
+        with_context :
+            If True, also consider the context of the Concepts for the
+            refinement.
+
+        Returns
+        -------
+        :
+            True if this Concept is a refinement of another Concept
+        """
+        if not isinstance(other, Concept):
+            raise TypeError(f"Cannot check refinement between {type(other)} and Concept")
```

Comment:
I think the logic here should be to just return False instead of throwing an error.